model/post.go

```diff
@@ -717,3 +720,10 @@ func (o *Post) IsFromOAuthBot() bool {
 	props := o.GetProps()
 	return props["from_webhook"] == "true" && props["override_username"] != ""
 }
+
+func (o *Post) ToNilIfInvalid() *Post {
+	if o.Id == "" {
+		return nil
+	}
+	return o
+}
```

Comment:
@sbishel In the [thread store](https://github.com/mattermost/mattermost-server/blob/36b11603f1acf82d38bfbfe95dd794aff92f25ae/store/sqlstore/thread_store.go#L310), we have to do a left join between threads and posts, and some of the posts might be nil because it is valid for a thread to exist without the root post.

Not elegant, I agree, but I couldn't think of anything better...